packages/flame/lib/src/components/sprite_component.dart

```diff
@@ -15,12 +15,50 @@ export '../sprite.dart';
 class SpriteComponent extends PositionComponent
     with HasPaint
     implements SizeProvider {
+  /// When set to true, the component is auto-resized to match the
+  /// size of underlying sprite.
+  late bool _autoResize;
+
+  /// Returns current value of auto resize flag.
+  bool get autoResize => _autoResize;
+
+  /// Sets the given value of autoResize flag. Will update the [size]
+  /// to fit srcSize of [sprite] if set to  true.
+  set autoResize(bool value) {
+    _autoResize = value;
+    if (value) {
+      if (_sprite != null) {
+        size.setFrom(_sprite!.srcSize);
+      } else {
+        size.setZero();
```

Comment:
Shouldn't we just do nothing when it is false? I would expect the previous size to still be there if setting auto size to false

Reply:
Wouldn't that break the contract of this setter? If `autoResize` is on and there is no sprite, wouldn't you expect the size of component to be zero as well?

Reply:
Ah, got ya, you are only setting the zero when there is no sprite. That makes sense!!!